reconversion to H2 chains invalid when supply country == demand country

The green chains that end with ` + reconv. to H2` are invalid when `country==region`. See example below. I don't know if this is semantically intended, but it may have implications for the blue chains, where production is "domestic". 

Here is an example:
```python
api.calculate(
    chain="Methane (SOEC) + reconv. to H2",
    country="Spain",
    output_unit="USD/MWh",
    region="Spain",
    res_gen="Wind-PV-Hybrid",
    scenario="2040 (medium)",
    secproc_co2="Direct Air Capture",
    secproc_water="Sea Water desalination",
    ship_own_fuel=False,
    transport="Pipeline",
)
```
Output:

```python
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[2], line 1
----> 1 api.calculate(
      2     chain="Methane (SOEC) + reconv. to H2",
      3     country="Spain",
      4     output_unit="USD/MWh",
      5     region="Spain",
      6     res_gen="Wind-PV-Hybrid",
      7     scenario="2040 (medium)",
      8     secproc_co2="Direct Air Capture",
      9     secproc_water="Sea Water desalination",
     10     ship_own_fuel=False,
     11     transport="Pipeline",
     12 )

File c:\Users\user\ptx-boa\ptxboa\api.py:198, in PtxboaAPI.calculate(self, scenario, secproc_co2, secproc_water, chain, res_gen, region, country, transport, ship_own_fuel, output_unit, user_data, optimize_flh, use_user_data_for_optimize_flh, tool_version_color)
    195 if transport not in TransportValues:
    196     logger.error(f"Invalid choice for transport: {transport}")
--> 198 data = data_handler.get_calculation_data(
    199     secondary_processes={
    200         "H2O-L": (
    201             DataHandler.get_dimensions_parameter_code(
    202                 "secproc_water", secproc_water
    203             )
    204             if secproc_water
    205             else None
    206         ),
    207         "CO2-G": (
    208             DataHandler.get_dimensions_parameter_code(
    209                 "secproc_co2", secproc_co2
    210             )
    211             if secproc_co2
    212             else None
    213         ),
    214     },  # type:ignore
    215     chain_name=chain,
    216     process_code_res=DataHandler.get_dimensions_parameter_code(
    217         "res_gen", res_gen
    218     ),  # type:ignore
    219     source_region_code=DataHandler.get_dimensions_parameter_code(
    220         "region", region
    221     ),  # type:ignore
    222     target_country_code=DataHandler.get_dimensions_parameter_code(
    223         "country", country
    224     ),  # type:ignore
    225     use_ship=(transport == "Ship"),
    226     ship_own_fuel=ship_own_fuel,
    227     optimize_flh=optimize_flh,
    228     use_user_data_for_optimize_flh=use_user_data_for_optimize_flh,
    229 )
    231 _values, result_df = PtxCalc.calculate(data)
    233 # conversion to output unit

File c:\Users\user\ptx-boa\ptxboa\api_data.py:828, in DataHandler.get_calculation_data(self, secondary_processes, chain_name, process_code_res, source_region_code, target_country_code, use_ship, ship_own_fuel, optimize_flh, use_user_data_for_optimize_flh)
    788 def get_calculation_data(
    789     self,
    790     secondary_processes: Dict[FlowCodeType, ProcessCodeType],
   (...)
    798     use_user_data_for_optimize_flh: bool = False,
    799 ) -> CalculateDataType:
    800     """Create data for calculation.
    801 
    802     Parameters
   (...)
    826         _description_
    827     """
--> 828     data = self._get_calculation_data(
    829         secondary_processes=secondary_processes,
    830         chain_name=chain_name,
    831         process_code_res=process_code_res,
    832         source_region_code=source_region_code,
    833         target_country_code=target_country_code,
    834         use_ship=use_ship,
    835         ship_own_fuel=ship_own_fuel,
    836         use_user_data=True,
    837     )
    839     # get optimizedFLH?
    840     if optimize_flh:
    841         # if we have user data BUT it should NOT be used for optimization
    842         # get a different dataset for optimization

File c:\Users\user\ptx-boa\ptxboa\api_data.py:942, in DataHandler._get_calculation_data(self, secondary_processes, chain_name, process_code_res, source_region_code, target_country_code, use_ship, ship_own_fuel, use_user_data)
    928     use_ship = True
    930 transport_distances = self._get_transport_distances(
    931     source_region_code,
    932     target_country_code,
   (...)
    938     existing_pipeline_cap,
    939 )
    941 chain_steps_main_export, chain_steps_transport, chain_steps_main_import = (
--> 942     self._filter_chain_processes(chain, transport_distances)
    943 )
    945 used_flows_main_chain: set[FlowCodeType] = set()
    946 used_flows_main_export: set[FlowCodeType] = set()

File c:\Users\user\ptx-boa\ptxboa\api_data.py:1142, in DataHandler._filter_chain_processes(cls, chain, transport_distances)
   1139         result_main_import.append(process_step)
   1141 # CHECK that flow chain is correct
-> 1142 cls._validate_process_chain(
   1143     [
   1144         chain[p]
   1145         for p in result_main_export + result_transport + result_main_import
   1146     ],
   1147     chain["FLOW_OUT"],
   1148 )
   1150 return result_main_export, result_transport, result_main_import

File c:\Users\user\ptx-boa\ptxboa\api_data.py:1089, in DataHandler._validate_process_chain(cls, process_codes, final_flow_code)
   1087     flow_code = process["main_flow_code_out"]  # type:ignore
   1088 if flow_code != final_flow_code:
-> 1089     raise AssertionError(
   1090         f"flow_code != flow_code_in: {flow_code} != {final_flow_code}"
   1091     )

AssertionError: flow_code != flow_code_in: CH4-G != H2-G
```

--- a/app/sidebar.py
+++ b/app/sidebar.py
@@ -175,16 +175,37 @@ def main_settings_blue(api: PtxboaAPI):
         key="country",
     )
 
-    production_location = st.radio(
-        "Where does production take place:",
-        ["supply", "demand"],
-        horizontal=True,
-        format_func=lambda x: {
-            "supply": "Supply Country",
-            "demand": "Demand Country",
-        }.get(x, x),
-        help=read_markdown_file("md/helptext_sidebar_blue_ptx_production_country.md"),
-    )
+    if st.session_state["region"] != st.session_state["country"]:
+        production_location = st.radio(
+            "Where does production take place:",
+            ["supply", "demand"],
+            horizontal=True,
+            format_func=lambda x: {
+                "supply": "Supply Country",
+                "demand": "Demand Country",
+            }.get(x, x),
+            help=read_markdown_file(
+                "md/helptext_sidebar_blue_ptx_production_country.md"
+            ),
+            key="production_location_radio1",
+        )
+    else:
+        # necessary to make a new st.radio to reset selected value to "supply"
+        production_location = st.radio(
+            "Where does production take place:",
+            ["supply", "demand"],
+            index=0,
+            horizontal=True,
+            format_func=lambda x: {
+                "supply": "Supply Country",
+                "demand": "Demand Country",
+            }.get(x, x),
+            help=read_markdown_file(
+                "md/helptext_sidebar_blue_ptx_production_country.md"
+            ),
+            key="production_location_radio2",
+            disabled=True,
+        )
 
     product_labels = {
         "CHX-L": "FT e-fuels",
@@ -283,7 +304,11 @@ def main_settings_blue(api: PtxboaAPI):
         index=0,
     )
 
-    if product == "H2-G" and production_location == "supply":
+    if (
+        product == "H2-G"
+        and production_location == "supply"
+        and st.session_state["region"] != st.session_state["country"]
+    ):
         nh3_transport = st.toggle(
             "Transport NH₃ and reconvert to H₂",
             value=False,

